.github/workflows/solidity.yml

```diff
@@ -7,6 +7,20 @@ env:
   NODE_OPTIONS: --max_old_space_size=5120
 
 jobs:
+  changes:
+    name: Detect changes
+    runs-on: ubuntu-latest
+    outputs:
+      changes: ${{ steps.changes.outputs.src }}
+    steps:
+      - uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721 # v2.10.2
+        id: changes
+        with:
+          filters: |
+            src:
+              - 'contracts/**/*'
+              - '.github/workflows/solidity.yml'
+
   split-tests:
```

Comment:
Don't forget to add:
` needs: changes`

On all jobs that require `outputs.changes`